nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/AbstractCSVRecordReader.java

```diff
@@ -158,4 +180,46 @@ protected String trim(String value) {
     public RecordSchema getSchema() {
         return schema;
     }
+
+    /**
+     * This method searches using the specified Reader character-by-character until the
+     * record separator is found.
+     * @param reader the Reader providing the input
+     * @param recordSeparator the String specifying the end of a record in the input
+     * @throws IOException if an error occurs during reading, including not finding the record separator in the input
+     */
+    protected void readNextRecord(Reader reader, String recordSeparator) throws IOException {
+        int indexIntoSeparator = 0;
+        int recordSeparatorLength = recordSeparator.length();
+        int code = reader.read();
+        while (code != -1) {
+            char nextChar = (char)code;
+            if (recordSeparator.charAt(indexIntoSeparator) == nextChar) {
+                if (++indexIntoSeparator == recordSeparatorLength) {
+                    // We have matched the separator, return the string built so far
+                    return;
+                }
```

Comment:
@mattyb149 @exceptionfactory A concern I have with this logic is what happens when the record separator is escaped in the data? How will you tell whether you have an end of record or an escaped record separator.  I did a quick look on line and found that a new line character which usually is the end of a CSV record can be embedded in the data.

Reply:
I'll look at capturing escape characters, maybe some sort of flag indicating that we're in an escape sequence. Good catch!

Reply:
Actually come to think of it, this capability is to skip lines aren't AREN'T valid CSV, otherwise you can use SampleRecord. Because we aren't assuming the skipped rows to be valid CSV, we should be able to just match on the newline character. I'll put in some code anyway but what are your thoughts?